### PR TITLE
Add tutorial notebook for RISE on FRB timeseries data

### PR DESCRIPTION
**Pull request by loostrum**
Also includes the FRB data itself, so this PR relates to both #707 and #710.

The FRB data and model nearly work with DIANNA timeseries as-is. The only issue is that the data needs some reshaping, as DIANNA expects data in (time, channel) format while the FRB model uses (channel, time). This is further explained in the notebook.

### CODE REVIEW

**Comment by loostrum** [comment]
To Do:

- add to README in tutorials folder (need to find a logo to represent FRB data)

**Comment by loostrum** [comment]
Tutorial added to README, logo will come later (see #719).
I've also added an explainer on a few modified versions of the input, to make the notebook a bit more interesting.

**Comment by loostrum** [comment]
#718 is handled in this PR as well - I considered doing it separately but when reviewing notebooks I thought it easier to just have everything in one go.

**Comment by elboyran** [comment]
Ah, meanwhile, I have messed up with the structure affecting mist the notebooks and their data, labels, and models in #699 (currently up to reviewing)... I wonder what order of merging is best?

File: tutorials/rise_timeseries_frb.ipynb [hunk 1]
```diff
@@ -0,0 +1,506 @@
+{
```
Comment (elboyran) [inline tutorials/rise_timeseries_frb.ipynb:1]: <p>I would put the paper name and an underlying link. </p><p>Also would add link to the RISE paper (just linked to the RISE abbreviation): <a href="http://bmvc2018.org/contents/papers/1064.pdf" target="_blank">http://bmvc2018.org/contents/papers/1064.pdf</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dianna-ai/dianna/pull/714/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

File: tutorials/README.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ Run the tutorials directly in Google Colab by clicking on the Colab buttons belo
...
 | | | [<img width="25" alt="Coffe Logo" src="https://user-images.githubusercontent.com/3244249/241999275-9ab50a0f-5da3-41d2-80e9-70d2c8769162.jpg">](lime_timeseries_coffee.ipynb) or [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/dianna-ai/dianna/blob/main/tutorials/lime_timeseries_coffee.ipynb) | |
+| | | [FRB](rise_timeseries_frb.ipynb) or [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/dianna-ai/dianna/blob/main/tutorials/rise_timeseries_frb.ipynb) | |
```
Comment (cwmeijer) [inline tutorials/README.md:26]: It would be great to add some image/icon for the dataset.
There are some nice pics on https://en.wikipedia.org/wiki/Fast_radio_burst
Some image of aperatif would be great as well, or just some deep space pic.
I think you should be able to find something with the right license.

File: .gitignore [hunk 3]
```diff
@@ -17,6 +17,9 @@ coverage.xml
 .tox
 *word_vectors.txt.pt
 
+# tutorial model that is downloaded automatically
```
Comment (cwmeijer) [inline .gitignore:20]: good idea!

File: tutorials/rise_timeseries_frb.ipynb (hunk 1 above)
Comment (elboyran) [inline tutorials/rise_timeseries_frb.ipynb:1]: <p>Would it be useful to add 1-2 sentences explaining what is seen in the data?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dianna-ai/dianna/pull/714/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (elboyran) [inline tutorials/rise_timeseries_frb.ipynb:1]: <p>Add a sentence above again on what classification task is.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dianna-ai/dianna/pull/714/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (elboyran) [inline tutorials/rise_timeseries_frb.ipynb:1]: <p>Maybe the model deserves a separate small section with 1/2 sentences above this one?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dianna-ai/dianna/pull/714/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (elboyran) [inline tutorials/rise_timeseries_frb.ipynb:1]: <p>1 line explaining what that is?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dianna-ai/dianna/pull/714/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

**Review by cwmeijer — CHANGES_REQUESTED** [review]
Great notebook with the extra's about shifting and adding other sources. One of the coolest tutorials that we have now! Thanks for giving this some love ;-)

I made some comments about some details.

**Comment by loostrum** [comment]
Thanks for your comments @elboyran and @cwmeijer, I'll get back to them early next week!

**Comment by elboyran** [comment]
> Thanks for your comments @elboyran and @cwmeijer, I'll get back to them early next week!

@loostrum , at the last stand-up, only myself and @cwmeijer were present and I had put a "stand-up" label for my issue (and branch) #667 (PR #699). We think, that if you have to address some comments anyway, maybe you can also apply the small changes according to the new folder structure and we merge my PR first, what do you think? I'll put soon the folder structure in "main" and in "667-reorganize" for convenience.

**Comment by elboyran** [comment]
Old and proposed folder structures of `dianna/dianna `and `dianna/tutorials` :
---------------------
## `main` branch
### `dianna` (6 directories, 56 files)

├── __init__.py
├── ...
├── dashboard
│   ├── ...
├── data
│   ├── bee_2.png
│   ├── binary-mnist.npz
│   ├── digit0.jpeg
│   ├── digit0.jpg
│   ├── digit0.png
│   ├── digit1.png
│   ├── labels_mnist.txt
│   ├── labels_resnet50.txt
│   ├── labels_text.txt
│   ├── leafsnap_model.onnx
│   ├── logo.png
│   ├── mnist_model_tf.onnx
│   ├── movie_review_model.onnx
│   ├── movie_reviews_word_vectors.txt
│   ├── season_prediction_model_temp_max_binary.onnx
│   ├── weather_data.npy
│   └── weather_data_labels.txt
├── methods
│   ├── ...
├── utils
│   ├── ...
└── visualization
│   ├── ...

### `tutorials` (4 directories, 38 files)

├── README.md
├── conversion_onnx
│   ├── ...
├── data
│   ├── binary-mnist.npz
│   ├── coffee_test.csv
│   ├── coffee_train.csv
│   ├── leafsnap_classes.csv
│   ├── leafsnap_example_acer_rubrum.jpg
│   ├── movie_reviews_word_vectors.txt
│   └── shapes.npz
├── demo.ipynb
├── img
│   ├── bee.jpg
│   └── digit0.jpg
├── kernelshap_geometric_shapes.ipynb
├── kernelshap_mnist.ipynb
├── kernelshap_tabular_penguin.ipynb
├── kernelshap_tabular_weather.ipynb
├── lime_images.ipynb
├── lime_tabular_penguin.ipynb
├── lime_tabular_weather.ipynb
├── lime_text.ipynb
├── lime_timeseries_coffee.ipynb
├── lime_timeseries_weather.ipynb
├── models
│   ├── coffee.onnx
│   ├── geometric_shapes_model.onnx
│   ├── leafsnap_model.onnx
│   ├── mnist_model.onnx
│   ├── mnist_model_tf.onnx
│   ├── movie_review_model.onnx
│   ├── penguin_model.onnx
│   ├── season_prediction_model_temp_max_binary.onnx
│   └── sunshine_hours_regression_model.onnx
├── rise_imagenet.ipynb
├── rise_mnist.ipynb
├── rise_text.ipynb
└── rise_timeseries_weather.ipynb

----------------------------
## `667-reorganize`  branch
### `dianna` (8 directories, 66 files)
├── __init__.py
├── ...
├── dashboard
│   ├── ...
├── data
│   ├── bee.jpg
│   ├── binary-mnist.npz
│   ├── digit0.jpeg
│   ├── digit0.jpg
│   ├── digit0.png
│   ├── digit1.png
│   ├── leafsnap_example_acer_rubrum.jpg
│   ├── logo.png
│   ├── shapes.npz
│   └── weather_data.npy
├── labels
│   ├── coffee_test.csv
│   ├── coffee_train.csv
│   ├── labels_mnist.txt
│   ├── labels_resnet50.txt
│   ├── labels_text.txt
│   ├── leafsnap_classes.csv
│   ├── movie_reviews_word_vectors.txt
│   └── weather_data_labels.txt
├── methods
│   ├── ...
├── models
│   ├── coffee.onnx
│   ├── geometric_shapes_model.onnx
│   ├── leafsnap_model.onnx
│   ├── mnist_model.onnx
│   ├── mnist_model_tf.onnx
│   ├── movie_review_model.onnx
│   ├── penguin_model.onnx
│   ├── season_prediction_model_temp_max_binary.onnx
│   └── sunshine_hours_regression_model.onnx
├── utils
│   ├── ...
└── visualization
    ├── ...

### `tutorials` (5 directories, 20 files)

├── README.md
├── conversion_onnx
│   ├── keras2onnx.ipynb
│   ├── pytorch2onnx.ipynb
│   ├── skl2onnx.ipynb
│   └── tensorflow2onnx.ipynb
├── demo.ipynb
└── explainers
    ├── KernelSHAP
    │   ├── kernelshap_geometric_shapes.ipynb
    │   ├── kernelshap_mnist.ipynb
    │   ├── kernelshap_tabular_penguin.ipynb
    │   └── kernelshap_tabular_weather.ipynb
    ├── LIME
    │   ├── lime_images.ipynb
    │   ├── lime_tabular_penguin.ipynb
    │   ├── lime_tabular_weather.ipynb
    │   ├── lime_text.ipynb
    │   ├── lime_timeseries_coffee.ipynb
    │   └── lime_timeseries_weather.ipynb
    └── RISE
        ├── rise_imagenet.ipynb
        ├── rise_mnist.ipynb
        ├── rise_text.ipynb
        └── rise_timeseries_weather.ipynb

**Comment by loostrum** [comment]
> > Thanks for your comments @elboyran and @cwmeijer, I'll get back to them early next week!
> 
> @loostrum , at the last stand-up, only myself and @cwmeijer were present and I had put a "stand-up" label for my issue (and branch) #667 (PR #699). We think, that if you have to address some comments anyway, maybe you can also apply the small changes according to the new folder structure and we merge my PR first, what do you think? I'll put soon the folder structure in "main" and in "667-reorganize" for convenience.

Yes that seems like a good idea! When #699 is merged I'll check that this PR then matches the new layout

File: tutorials/rise_timeseries_frb.ipynb (hunk 1 above)
Comment (loostrum) [reply tutorials/rise_timeseries_frb.ipynb:1]: <p>The linked paper is the RISE paper, this part of the introduction is actually the same for all rise tutorials we have. I'll make a new issue to fix it in all of them.</p><p> </p>

Comment (loostrum) [reply tutorials/rise_timeseries_frb.ipynb:1]: <p>Good idea, I've added an explanation</p>

Comment (loostrum) [reply tutorials/rise_timeseries_frb.ipynb:1]: <p>Done</p>

Comment (loostrum) [reply tutorials/rise_timeseries_frb.ipynb:1]: <p>I've added this information to the classification task section</p>

Comment (loostrum) [reply tutorials/rise_timeseries_frb.ipynb:1]: <p>It can be anything really, but I can mention a few examples</p><p> </p>

Comment (loostrum) [reply tutorials/rise_timeseries_frb.ipynb:1]: <p>I see actually that the section I wrote on trying both the shift and a deleterious signal has disappeared :/ I'll rewrite that...</p>

File: tutorials/README.md [hunk 4]
```diff
@@ -39,7 +40,4 @@ The datasets used in the tutorials are represented with their respective logos:
 |*Tabular*| [Penguin dataset](https://www.kaggle.com/code/parulpandey/penguin-dataset-the-new-iris)| <img width="75" alt="Penguin Logo" src="https://github.com/dianna-ai/dianna/assets/3244249/c7716ad3-f992-4557-80d9-1d8178c7ed57"> | |
 |           | [Weather dataset](https://zenodo.org/record/7525955) | <img width="25" alt="Weather Logo" src="https://github.com/dianna-ai/dianna/assets/3244249/3ff3d639-ed2f-4a38-b7ac-957c984bce9f">|
 
-The ONNX models used in the tutorials are available at [tutorials/models](https://github.com/dianna-ai/dianna/tree/main/tutorials/models).
-
-
-
+The ONNX models used in the tutorials are available at [tutorials/models](https://github.com/dianna-ai/dianna/tree/main/tutorials/models). The FRB model is available [here](https://zenodo.org/record/10656614)
```
[the remark this answers is not in the document]
Comment (loostrum) [reply tutorials/README.md:43]: Good point, I wasn't sure what to do as the statement "the models are available here" isn't true for the FRB case. Should we add a general remark that they're either available on our github or linked from the notebook?

File: tutorials/README.md (hunk 2 above)
Comment (loostrum) [reply tutorials/README.md]: I made a tiny version of the Apertif logo for this :)

File: tutorials/rise_timeseries_frb.ipynb (hunk 1 above)
Comment (loostrum) [reply tutorials/rise_timeseries_frb.ipynb:1]: See issue #724

**Comment by loostrum** [comment]
Note: the Mac OS on Python 3.11 CI is failing for reasons unrelated to this PR

File: tutorials/rise_timeseries_frb.ipynb [hunk 5]
```diff
@@ -0,0 +1,545 @@
...
```
Comment (elboyran) [inline tutorials/rise_timeseries_frb.ipynb:1]: <p>I am not sure I am seeing the notebook correctly in this viewer, but if so, this is partly an exact repetition of a paragraph below. Copy&amp; Paste lapse?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dianna-ai/dianna/pull/714/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (elboyran) [inline tutorials/rise_timeseries_frb.ipynb:1]: <p>Small detail: you consider 2 modifications and then you address them in reverse order ;-) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dianna-ai/dianna/pull/714/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (elboyran) [inline tutorials/rise_timeseries_frb.ipynb:1]: <p>Were you planning to add more disturbing signals until the model cracks?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/dianna-ai/dianna/pull/714/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

**Review by elboyran — COMMENTED** [review]
Small suggestions in notebook.

File: tutorials/rise_timeseries_frb.ipynb (hunk 5 above)
Comment (loostrum) [reply tutorials/rise_timeseries_frb.ipynb:1]: <p>I did, but the output figure weren't stored apparently. Fixed now.</p>

Comment (loostrum) [reply tutorials/rise_timeseries_frb.ipynb:1]: <p>Good catch! Fixed :)</p><p> </p>

Comment (loostrum) [reply tutorials/rise_timeseries_frb.ipynb:1]: <p>Yep that's a copy-paste mistake, fixed now</p><p> </p>

**Comment by loostrum** [comment]
@elboyran could you check if your comments are now adequately addressed? Then I think this is ready for merging

**Review by elboyran — APPROVED** [review]
Looks nice. Only some MD links checks should pass.
